### PR TITLE
CONSOLE-2527: Sprint 198 translations

### PR DESCRIPTION
**Pull request by rebeccaalpert**
Added latest round of translation files. I went through and updated some issues, detailed below by file.

I'll bug the translation team about the translation stuff. I fixed the easy buggy ones locally. We need the team to update Memsource or our changes will keep getting overwritten.

#### Japanese:
console-app.json 
* "VolumeSnapshot": "VolumeSnapshot", but "VolumeSnapshots": "ボリュームスナップショット",

volume-snapshot-class-details.json
* "VolumeSnapshotClass details": "ボリュームスナップショットの詳細", - VolumeSnapshotClass is left untranslated in console-app.json; this appears to be the VolumeSnapshot translation rather than VolumeSnapshotClass

frontend/packages/console-app/locales/ja/volume-snapshot-content.json
* "VolumeSnapshotContent details": "ボリュームスナップショットコンテンツの詳細", but it’s "VolumeSnapshotContent": "VolumeSnapshotContent", in console-app.json and  "VolumeSnapshotContent": "VolumeSnapshotContent", in same file

olm.json
* "InstallPlan failed": "InstalPlan に失敗しました", (We have brought this up at least twice now and it's still not fixed)

#### Korean:
console-app.json
* "VolumeSnapshot details": "볼륨 스냅샷 세부 정보", but  "VolumeSnapshot": "VolumeSnapshot",

volume-snapshot-class-details.json
*  "VolumeSnapshotClass details": "볼륨 스냅샷 클래스 세부 정보", but "VolumeSnapshotClass": "VolumeSnapshotClass", in console-app.json and volume-snapshot-details.json

volume-snapshot.json
* "VolumeSnapshotClass": "볼륨 스냅샷 클래스", but  "VolumeSnapshotClass": "VolumeSnapshotClass" in volume-snapshot-class-details.json

devconsole.json
* "Build Configs": "빌드 설정", but "Select a Project to view the list of BuildConfigs or <2>create a Project</2>.": "프로젝트를 선택하여 BuildConfigs 목록을 보거나 <2>프로젝트를 생성</2>합니다.", and "Select a Project to view the list of BuildConfigs": "빌드 설정 목록을 표시할 프로젝트를 선택",

#### Chinese:
console-app.json
* "Create VolumeSnapshot": "创建卷快照", but "VolumeSnapshot": "VolumeSnapshot",
* "VolumeSnapshotClass with same provisioner as claim": "带有与声明相同的置备程序的卷快照类", but "VolumeSnapshotClass": "VolumeSnapshotClass",
* "VolumeSnapshotContent details": "卷快照内容详情", but "VolumeSnapshotContent": "VolumeSnapshotContent",
* "VolumeSnapshot details": "卷快照详情", but "VolumeSnapshot": "VolumeSnapshot",

nodes.json
* "Mark as unschedulable": "将节点标记为不可调度。", - still has period added (We have brought this up at least twice now and it's still not fixed)

volume-snapshot-class-details.json
* "VolumeSnapshotClass details": "卷快照类详情", but "VolumeSnapshotClass": "VolumeSnapshotClass",

volume-snapshot-content.json
* "VolumeSnapshotContent details": "卷快照内容详情", but "VolumeSnapshotContent": "VolumeSnapshotContent",

volume-snapshot-details.json
* "VolumeSnapshot details": "卷快照详情", but "VolumeSnapshot": "VolumeSnapshot",

olm.json
`"**Operator backed** includes a variety of services managed by Kubernetes controllers.": "**Ooperator 支持的** 包括由 Kubernetes 控制器管理的各种服务。",` - typo in translation

public.json
* "Home": "家", should be "Home": "主页", (We have brought this up at least twice now and it's still not fixed)
* "PersistentVolumeClaim details": "持久性卷声明详情", but "PersistentVolumeClaim": "PersistentVolumeClaim",
* "Create StorageClass": "创建存储类", but "StorageClass": "StorageClass", and "StorageClass details": "存储类详情",

details-page.json
* "Edit annotations": "Pod 注解", should be "Edit annotations": "编辑注解", (We have brought this up at least twice now and it's still not fixed)

modal.json
* "Edit annotations": "Pod 注解", should be "Edit annotations": "编辑注解", (We have brought this up at least twice now and it's still not fixed)

persistent-volume-claim.json 
* "PersistentVolumeClaim details": "持久性卷声明详情", but "PersistentVolumeClaims": "PersistentVolumeClaims" and "PersistentVolumeClaim": "PersistentVolumeClaim",

persistent-volume.json
* "PersistentVolume details": "持久性卷", but "PersistentVolumes": "PersistentVolumes"

devconsole.json
* "Build Configs": "构建配置", but "Select a Project to view the list of BuildConfigs or <2>create a Project</2>.": "选择一个项目来查看 BuildConfigs 列表或<2>创建一个项目</2>。",
* "Dockerfile": "Docker", (We have brought this up at least twice now and it's still not fixed)

### Text updates
ceph-storage-plugin.json  
* Updated  "Description(Optional)" key to  "Description (Optional)"
* "Mirroring Tier - list of spread-layers async-mirroring to all mirrors with locality optimization (will allocate on the closest region to the source endpoint) mirroring requires at least two backing stores." - Added period before "mirroring"
* "Full duplication of the data in each chosen resource By default a replica of one copy per location is used. includes failure tolerance in case of resource failure." - Capitalized "Includes" and added a period to the middle of the sentence.
* "Spreading the data across the chosen resources does not includes failure tolerance in case of resource failure." - Updated "includes" to "include"
* "This card shows the used capacity for Usable storage, broken-down by different kubernetes resources." Appears to no longer be used in console; applied no fixes
* ""Are you sure you want to replace <1>{{diskName}}</1> ?" - removed space
* "Where can I find google cloud credentials?" - capitalized Google Cloud
* "Upload a .json file with the service account keys provided by google cloud storage." - capitalized Google Cloud
*  " Object Bucket Claim Name" - translation appears to no longer be in use; did not update those, but updated English file
* "Download <1>{{SCRIPT_NAME}}</1> script and run on the RHCS cluster, then upload the results(JSON) in the External cluster metadata field." - added space before JSON in translations; English is correct
* "An encryption key will be generated for each persistent volume (block) created using an encryption enabled storageclass." - storage class to StorageClass
* "An encryption key will be generated for each persistent volume created using this storageclass." - storageclass to StorageClass
* ‘By enabling thick-provisioning, volumes will allocate the requested capacity upon volume creation. Volume creation to be slower when thick-provisioning is enabled." - updated "to be" to "will be"
* "A Storage pool is a logical entity providing elastic capacity to applications and workloads. Pools provide a means of supporting policies for access data resilience and storage efficiency." - string no longer in use; made no updates
* "A new storage class will be created with encryption enabled. Encryption key for each Persistent volume (block only) will be generated." - string no longer in use; made no updates
* "An encryption key for each Persistent volume (block only) will be generated." - string no longer in use; made no updates
* "An MCG Buckets data location is determined by a policy called a Bucket Class" - string no longer in use; made no updates
* "A list of all MCG (Multicloud Object Gateway) resources that are currently in use. Those resources are used to store data according to the buckets policies and can be a cloud-based resource or a bare metal resource." - string no longer in use; made no updates

devconsole.json
* "Select the checkbox 'Create a route to the application' to edit advanced routing options" - Updated quotation marks
* "Port must be an Integer.": to "Port must be an integer"

kubevirt-plugin.json 
* "For customizing boot source, select 'Customize boot source' from the template actions menu." - updated quotes
* "Cancel customization ?" - removed space
* "This action will delete the boot source copy as well as all other temporary resources such as  the virtual machine that is running the console." - removed extra space
* "The original template will be available for virtual machines creation and remain intact through the entire process except when boot source is being cloned." - added "the" 
* "for": " ", - key appears to have been removed; not updating anything here
* "We haven't detected authorized key for the SSH access. SSH access will be enabled without authorized key" - Updated to "We haven't detected an authorized key for SSH access. SSH access will be enabled without an authorized key" and removed backticks
* "<0>Remember authorized SSH key</0><1>Store this key in a namespaced secret and suggest it next time creating a VM</1><2>The key will be stored after the machine will be created</2>" - Updated to "<0>Remember authorized SSH key</0><1>Store this key in a namespaced secret and suggest it the next time a VM is created</1><2>The key will be stored after the machine is created</2>"
* Boot source customization will apply to a boot source copy, saved on a new template. This template will be a clone of the original boot source template {{templateName}}. The customized boot source will be saved to the new template." - removed comma
* "This boot source is marked as CD-ROM, and requires allocated resources (disk) to customize it. Please allocate Persistent Volume Claim for customization process." - adding "a", "the", and dehumanized PVC
* "While customizing this template will not be available for virtual machine creation." - added comma
* "SSH Service can be managed in the VMs details page" to VM’s

lso-plugin.json
* "Selecting nodes allow you to limit the discovery for available disks to specific nodes." Added "s" to allow

noobaa-storage-plugin.json
* File is not being used; there are no source files in this folder. I am removing the translation files from this folder.

olm.json
* "This operator provides a custom interface you can include in your console.  Make sure you trust this operator before enabling its interface." - removed extra space
* "A Subscription for this Operator already exists in Namespace '{{namespace}}'"  - fixed quotes
* "Please note that installing non Red Hat operators into openshift namespaces and enabling monitoring voids user support. Enabling cluster monitoring for non Red Hat operators can lead to malicious metrics data overriding existing cluster metrics. For more information, see the <2>cluster monitoring documentation</2>. - fixed grammar
* "This operator will be able to provide a custom interface and run any kubernetes command as the logged in user.  Make sure you trust it before enabling." - removed extra space and capped K8s 

pipelines-plugin.json 
* "Select a Project to view the list of{{pipelineRunLabel}} or <4>create a Project</4>." - fixed spacing
* "Finally tasks", "Finally task,’ and "Add finally task’ - Final task(s)

### CODE REVIEW

**Review by afreen23 — APPROVED** [review]
ceph and lso looks good.
Thanks

**Comment by rebeccaalpert** [comment]
Rebased.

**Review by spadgett — APPROVED** [review]
/lgtm
/hold for approvals

@rebeccaalpert I really appreciate all the work you do to fix typos and mistakes in the strings. I think going forward, it would be better to track those items as Bugzilla bugs, particular as we automate translation drops. That way we can have the plugin teams who own the code test those changes, and we'll be able to fix those in separate PRs for easier review and verification.

**Comment by rebeccaalpert** [comment]
Rebased 
```
frontend/packages/kubevirt-plugin/locales/en/kubevirt-plugin.json
frontend/packages/kubevirt-plugin/src/components/ssh-service/SSHCreateService/SSHCreateService.tsx
frontend/packages/kubevirt-plugin/src/components/ssh-service/SSHPopover/SSHPopover.tsx
frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx
```

File: frontend/packages/console-app/locales/ja/volume-snapshot-content.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "VolumeSnapshotContent details": "ボリュームスナップショットコンテンツの詳細",
+  "Status": "ステータス",
+  "Size": "サイズ",
+  "VolumeSnapshot": "VolumeSnapshot",
+  "VolumeSnapshotContent": "VolumeSnapshotContent",
```
Comment (yanpzhan) [inline frontend/packages/console-app/locales/ja/volume-snapshot-content.json:6]: There is no "VolumeSnapshotContent" field on VolumeSnapshotContent detail page, the field should be "VolumeSnapshotClass". Refer to this bug: https://bugzilla.redhat.com/show_bug.cgi?id=1944590

**Comment by yanpzhan** [comment]
@rebeccaalpert 
Menus under "Storage" are not translated for Chinese/Korean.
Screenshot from 2021-05-12 12-52-29
Screenshot from 2021-05-12 12-48-45

There are many discordance for "VolumeSnapshot", "VolumeSnapshotClass","VolumeSnapshotContext", they are translated in "VolumeSnapshot details", "VolumeSnapshotClass details", "VolumeSnapshotContext details" and some detail page fields, but most of them are not translated. We should uniform their translations.
What's your opinion about these pages issue?

**Comment by rebeccaalpert** [comment]
@yanpzhan, yeah, the translators have been largely not translating those (except when they do!). I definitely agree they should be uniform. I would open a separate bug to track it. They haven't really been responsive on questions or comments lately, so it's good to have a paper trail of what the issues are.

File: frontend/packages/console-app/locales/ja/volume-snapshot-content.json (hunk 1 above)
Comment (rebeccaalpert) [reply frontend/packages/console-app/locales/ja/volume-snapshot-content.json:6]: It looks like this string is used in volume-snapshot-content-details.tsx and volume-snapshot-details.tsx.

**Comment by spadgett** [comment]
> @rebeccaalpert
> Menus under "Storage" are not translated for Chinese/Korean.
> Screenshot from 2021-05-12 12-52-29
> Screenshot from 2021-05-12 12-48-45
> 
> There are many discordance for "VolumeSnapshot", "VolumeSnapshotClass","VolumeSnapshotContext", they are translated in "VolumeSnapshot details", "VolumeSnapshotClass details", "VolumeSnapshotContext details" and some detail page fields, but most of them are not translated. We should uniform their translations.
> What's your opinion about these pages issue?

Hi @yanpzhan, can you open this as a separate bug? I agree we need to look at it, but it's an existing problem that's not introduced by the PR.

**Comment by yanpzhan** [comment]
@spadgett https://bugzilla.redhat.com/show_bug.cgi?id=1954540 has reported translation issues under storage menu, but it was closed as notabug, could you help to check if it could be reopened？

File: frontend/packages/console-app/locales/ja/volume-snapshot-content.json (hunk 1 above)
Comment (yanpzhan) [reply frontend/packages/console-app/locales/ja/volume-snapshot-content.json:6]: "VolumeSnapshotClass" is used in  volume-snapshot-content-details.tsx and and volume-snapshot-details.tsx. Not sure if it matters for translation.

**Comment by rebeccaalpert** [comment]
All set @yanpzhan; I reopened it.

File: frontend/packages/console-app/locales/ja/volume-snapshot-content.json (hunk 1 above)
Comment (rebeccaalpert) [reply frontend/packages/console-app/locales/ja/volume-snapshot-content.json:6]: I looked again and it looks like VolumeSnapshotContent is being used in volume-snapshot-details.tsx. Doesn't seem to be a bug to me, but let me know if there's something I'm missing.

**Comment by yanpzhan** [comment]
Thanks Rebecca. There is no big issue for existing translation files in the pr. I'd like to approve the pr so that all the translation files could take effect soon.

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**